cibuildwheel/linux.py

```diff
@@ -110,8 +110,8 @@ def build(options: BuildOptions) -> None:
             continue
 
         try:
-            with DockerContainer(docker_image, simulate_32_bit=platform_tag.endswith('i686')) as docker:
-                docker.copy_into(Path.cwd(), Path('/project'))
+            with DockerContainer(docker_image, simulate_32_bit=platform_tag.endswith('i686'), cwd='/project') as docker:
+                docker.copy_into(Path.cwd(), PurePath('/project'))
```

Comment:
Setting cwd to `/project` in the constructor feels a little strange, because I don't think `/project` has been created yet. I suppose Docker must create it if it's not there. But doesn't feel right to me. I think a more familar approach might be to have `docker.chdir(PurePath('/project'))` after the project is copied over. 

The cwd can be stored in DockerContainer as an ivar, and used whenever `call()` is called without a `cwd`. What do you think?

Reply:
Docker does document that passing `-w` will create the directory: https://docs.docker.com/engine/reference/commandline/create/#options

For the `DockerContainer`, I thought of it as just setting the "default directory" during the with-statement, and just adding an exception if necessary. And also that we then couldn't forget it, later.

But yes, `chdir` would also work, if you prefer that. I hadn't thought of that option. I guess we don't even need to store it, but could just `cd`, rather than storing it as an attribute?

Reply:
Ah, that's good that's it's documented. Then I think either approach is valid, actually. Happy to leave it as-is.

Reply:
Given that we don't have a `os.chdir` for macOS or Windows (and just use the working directory that's there at the start of cibuildwheel/`build`), I think the current approach might match better. But I'm happy changing it as well, if you prefer. I like the suggested `chdir` method as well (and we might end up needing it at some later point?). Your call!

@Czaki, any input on this? Do you prefer `with DockerContainer(..., cwd='/project'):` or `with DockerContainer(...) as docker: docker.chdir('/project')`?

Reply:
I prefer `with DockerContainer(..., cwd='/project'):` and change second argument of     
```
copy_into(self, from_path: Path, to_path: Optional[PurePath]=none)
    if to_path is None:
        to_path = self.cwd
    ...
```

Reply:
Oh, that's an interesting suggestion. I'll add that a as extra commit. If @joerick doesn't like it, it can just be removed before merging!

Reply:
Actually, that's not possible, @Czaki, because `self.cwd` can also be `None`. Do you suggest any way to fix it, or should we just keep `to_path` in `copy_into` mandatory?

Reply:
Raise exception with information that `cwd` should be set or `to_path`. 
I suggest this to reduce number of places where path is provided as literal.

Reply:
Yeah, possible, but ... it's still used in a lot of places, like `format(..., project='/project', ...)`, so doing this won't solve that all?